rpc/write.go

```diff
@@ -11,15 +11,15 @@ import (
 //   - invokeTxn: The invoke transaction to be added.
 //
 // Returns:
-//   - *AddInvokeTransactionResponse: the response of adding the invoke transaction
+//   - AddInvokeTransactionResponse: the response of adding the invoke transaction
 //   - error: an error if any
 func (provider *Provider) AddInvokeTransaction(
 	ctx context.Context,
 	invokeTxn *BroadcastInvokeTxnV3,
-) (*AddInvokeTransactionResponse, error) {
+) (AddInvokeTransactionResponse, error) {
 	var output AddInvokeTransactionResponse
 	if err := do(ctx, provider.c, "starknet_addInvokeTransaction", &output, invokeTxn); err != nil {
-		return nil, tryUnwrapToRPCErr(
+		return output, tryUnwrapToRPCErr(
```

Comment:
I think when there's an error, the function should return an empty struct value, not the potentially partially filled output variable.